### PR TITLE
FE-787 - Add Cypress integration tests for the Events feature

### PR DESCRIPTION
**Pull request by nicklemmon**
[FE-787](https://jira.int.messagesystems.com/browse/FE-787)

### What Changed
- Adds Cypress tests for Events list page and details page
- Parallelizes Cypress tests by breaking them up in to multiple groups using the `--group` flag and multiple Travis CI runners
- Removes remaining duplicate `/// <reference types="Cypress" />` for intelligent code completion in favor of setup file
- Updates page `<title>` for the Events list page and details page
- Replaces last remaining instances of local `stubRequest` functions with `cy.stubRequest()`

### How To Test
- Verify that Travis CI runs Cypress tests in two batches, reducing overall run time
- Verify that happy and unhappy paths fo the events page are adequately tested by Cypress

### To Do
- [ ] Incorporate feedback
- [x] Add tests for query params
- [x] Delete Jest tests duplicated by Cypress
- [x] Details Page - Add tests for message history clicking
- [x] Details Page - Add tests for disclaimer content rendering - **tested well with a Jest test**
- [x] Add tests for the orphaned events

### CODE REVIEW

File: src/pages/reports/messageEvents/components/SearchQuery.js [hunk 1]
```diff
@@ -9,18 +9,23 @@ import { eventsQuery } from 'src/helpers/validation';
 import { EVENTS_SEARCH_FILTERS } from 'src/constants';
```
Comment (shrirupa) [inline src/pages/reports/messageEvents/components/SearchQuery.js]: did anything else change here other than prettier ?

File: cypress/fixtures/events/message/200.get.1-result.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "subject": "Mock Subject 1",
+      "ip_pool": "things",
+      "sending_domain": "hotmail.com",
+      "rcpt_tags": ["abc", "def", "ghi"],
+      "type": "injection",
+      "raw_rcpt_to": "fake-recipient@hotmail.com",
+      "msg_from": "fake-sender@yahoo.com",
+      "recv_method": "esmtp",
+      "rcpt_to": "fake-recipient@hotmail.com",
+      "transmission_id": "21175596944301160",
```
Comment (shrirupa) [inline cypress/fixtures/events/message/200.get.1-result.json]: not necessary but can we replace this with simpler ids ?

File: cypress/tests/integration/events/listPage.spec.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    it('re-requests data when removing an active filter and when clicking "Clear All Filters"', () => {
+      cy.visit(PAGE_URL);
+
+      cy.stubRequest({
+        url: '/api/v1/events/message*',
+        fixture: 'events/message/200.get.different-results.json',
+      });
+
+      cy.findByLabelText('Filter by Recipient Email Address')
+        .type('hello@gmail.com')
```
Comment (shrirupa) [inline cypress/tests/integration/events/listPage.spec.js]: i think this should be different-recipient@hotmail.com.

File: cypress/tests/integration/events/listPage.spec.js [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+      it('renders documentation about each event type when hovering over a check box label', () => {
+        // NOTE: Rendering of these tooltips is a little unusual as Cypress does not have a .hover()
+        // that actually simulates user behavior. This means using the `mousover` event actually
+        // triggers a technically impossible application state when a real user is in control vs.
+        // simulated events
+        cy.findByText('Delay').trigger('mouseover');
+
+        cy.findByText('Remote MTA has temporarily rejected a message.').should('be.visible');
+
```
Comment (shrirupa) [inline cypress/tests/integration/events/listPage.spec.js]: we might need to tigger('mouseLeave') to have accurate screenshots
Screen Shot 2020-02-04 at 4 38 31 PM

File: cypress/tests/integration/events/listPage.spec.js [hunk 5]
```diff
@@ -0,0 +1,314 @@
+const PAGE_URL = '/reports/message-events';
+
+describe('The events page', () => {
+  beforeEach(() => {
```
Comment (shrirupa) [inline cypress/tests/integration/events/listPage.spec.js]: do you think we can add a test to check if the date filter works correctly ?

File: cypress/tests/integration/events/detailsPage.spec.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
+const PAGE_BASE_URL = '/reports/message-events/details';
```
Comment (shrirupa) [inline cypress/tests/integration/events/detailsPage.spec.js]: 🎉

File: cypress/tests/integration/events/listPage.spec.js [hunk 7]
```diff
@@ -0,0 +1,314 @@
+const PAGE_URL = '/reports/message-events';
+
```
Comment (shrirupa) [inline cypress/tests/integration/events/listPage.spec.js]: do we need the tests for forward backward and save as csv buttons at the bottom of the table ?

Comment (nicklemmon) [reply cypress/tests/integration/events/listPage.spec.js]: The pagination should be tested somewhere, you're. Let me take a stab at it.

Right now, testing the CSV download doesn't look possible until this gets resolved:
https://github.com/cypress-io/cypress/issues/949

That being said, there are some workarounds in that issue that _might_ work so I'm going to take a look at those as well.

File: cypress/tests/integration/events/listPage.spec.js (hunk 4 above)
Comment (nicklemmon) [reply cypress/tests/integration/events/listPage.spec.js]: Got it - added additional assertions that the tooltip should be invisible when not hovering but that behavior is broken due to a bug with Matchbox. I left a comment in the code describing the path forward.

File: src/pages/reports/messageEvents/components/SearchQuery.js (hunk 1 above)
Comment (nicklemmon) [reply src/pages/reports/messageEvents/components/SearchQuery.js]: I added new HTML `<label>` elements to resolve the a11y problem and make them easily accessible via `cy.findByLabelText()`!

**Comment by nicklemmon** [comment]
@shrirupa Ready for another pass when you have time.

**Pull request merged**